packages/settings/test/settings-spec.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  resetSettings,
+  getSettingsForDocument,
+  getDefaultSettings,
+  setGlobalSettings,
+  setSettingsForDocument,
+  hasSettingsForDocument,
+  clearDocumentSettings,
+  clearSettings,
+} from "../src/settings";
+import { expect } from "chai";
+
+context("settings utilities", () => {
+  beforeEach(() => {
+    resetSettings();
+  });
+
+  it("by default, deprecated and experimental suggestions will not be offered", () => {
+    const defaultSettings = getDefaultSettings();
+    expect(defaultSettings.codeAssist.deprecated).to.be.false;
+    expect(defaultSettings.codeAssist.experimental).to.be.false;
+  });
+
+  describe("getSettingsForDocument", () => {
+    it("will return the default settings when there are no settings for the document and no global settings", async () => {
+      const docSettings = await getSettingsForDocument("doc1");
+      expect(docSettings).to.equal(getDefaultSettings());
+    });
+
+    it("will return the global settings when there is are no settings for the document", async () => {
+      const globalSettings = {
+        codeAssist: { deprecated: true, experimental: false },
+        trace: { server: "off" as const },
+      };
+      setGlobalSettings(globalSettings);
+      const docSettings = await getSettingsForDocument("doc1");
+      expect(docSettings).to.equal(globalSettings);
```

Comment:
`to.deep.equal`?

because we don't care about **reference** equality here, we care about **value** equality.
Having reference equality is an implementation detail that may no longer apply if we add extra "defenses" to our settings module (deepClone).

Reply:
The comment above applies to all the tests in this file

Reply:
Done